src/main/java/io/gravitee/reporter/common/bulk/BulkProcessor.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright © 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.reporter.common.bulk;
+
+import io.gravitee.common.service.AbstractService;
+import io.gravitee.common.utils.RxHelper;
+import io.gravitee.reporter.api.Reportable;
+import io.gravitee.reporter.common.bulk.backpressure.BulkDropper;
+import io.gravitee.reporter.common.bulk.backpressure.FlowableBackPressureMemoryAware;
+import io.gravitee.reporter.common.bulk.compressor.BulkCompressor;
+import io.gravitee.reporter.common.bulk.compressor.CompressedBulk;
+import io.gravitee.reporter.common.bulk.compressor.GzipBulkCompressor;
+import io.gravitee.reporter.common.bulk.exception.NonRetryableException;
+import io.gravitee.reporter.common.bulk.sender.BulkSender;
+import io.gravitee.reporter.common.bulk.transformer.BulkTransformer;
+import io.gravitee.reporter.common.bulk.transformer.TransformedReport;
+import io.reactivex.rxjava3.annotations.NonNull;
+import io.reactivex.rxjava3.core.Completable;
+import io.reactivex.rxjava3.core.Maybe;
+import io.reactivex.rxjava3.disposables.Disposable;
+import io.reactivex.rxjava3.processors.UnicastProcessor;
+import io.reactivex.rxjava3.schedulers.Schedulers;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * @author Guillaume LAMIRAND (guillaume.lamirand at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+@Slf4j
+@RequiredArgsConstructor
+public class BulkProcessor extends AbstractService<BulkProcessor> {
+
+  public static final double RETRY_FACTOR = 1.5;
+
+  private final BulkSender bulkSender;
+  private final BulkConfiguration bulkConfiguration;
+  private final BulkTransformer bulkTransformer;
+  private final UnicastProcessor<Reportable> processor =
+    UnicastProcessor.create();
+  private BulkCompressor bulkCompressor = new GzipBulkCompressor();
+  private BulkDropper bulkDropper = new BulkDropper();
```

Comment:
Why do you create the dropper and the compressor here ? They will be passed as params when the processor will be instanciated no ?

Reply:
It was kind of default values I was using in CloudReporter, I can remove them and the annotation as well. wdyt ?